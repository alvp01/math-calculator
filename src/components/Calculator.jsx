import { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calcObj, setCalcObj] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (btnName) => {
    const newState = calculate(calcObj, btnName);
    setCalcObj(newState);
  };

  const displayValues = ({ total, next, operation }) => {
    if (total && !next && !operation) return `${total}`;
    if (!total && next && !operation) return `${next}`;
    if (total && !next && operation) return `${total} ${operation}`;
    if (total && next && operation) return `${next}`;
    return '0';
  };

  return (
    <div className="calculator-body">
      <div className="screen" data-testid="result">
        { displayValues(calcObj) }
      </div>
      <div className="keyboard">
        <div className="row">
          <button type="button" className="btn" onClick={() => handleClick('AC')}>AC</button>
          <button type="button" className="btn" onClick={() => handleClick('+/-')}>+/-</button>
          <button type="button" className="btn" onClick={() => handleClick('%')}>%</button>
          <button type="button" className="btn" onClick={() => handleClick('รท')}>รท</button>
        </div>
        <div className="row">
          <button type="button" className="btn" onClick={() => handleClick('7')}>7</button>
          <button type="button" className="btn" onClick={() => handleClick('8')}>8</button>
          <button type="button" className="btn" onClick={() => handleClick('9')}>9</button>
          <button type="button" className="btn" onClick={() => handleClick('x')}>x</button>
        </div>
        <div className="row">
          <button type="button" className="btn" onClick={() => handleClick('4')}>4</button>
          <button type="button" className="btn" onClick={() => handleClick('5')}>5</button>
          <button type="button" className="btn" onClick={() => handleClick('6')}>6</button>
          <button type="button" className="btn" onClick={() => handleClick('-')}>-</button>
        </div>
        <div className="row">
          <button type="button" className="btn" onClick={() => handleClick('1')} key='1'>1</button>
          <button type="button" className="btn" onClick={() => handleClick('2')}>2</button>
          <button type="button" className="btn" onClick={() => handleClick('3')}>3</button>
          <button type="button" className="btn" onClick={() => handleClick('+')}>+</button>
        </div>
        <div className="row last-row">
          <button type="button" className="btn" onClick={() => handleClick('0')}>0</button>
          <button type="button" className="btn" onClick={() => handleClick('.')}>.</button>
          <button type="button" className="btn" onClick={() => handleClick('=')}>=</button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
