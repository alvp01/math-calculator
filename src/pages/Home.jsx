import React from 'react';
import style from './Home.module.css';

const Home = () => (
  <div className={style.contentContainer}>
    <h1>Welcome to our page!</h1>
    <div className={style.paragraphs}>
      <p>
        {`Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
        incididunt ut labore et dolore magna aliqua. 
        Ut enim ad minim veniam, quis nostrud exercitation ullamco 
        laboris nisi ut aliquip ex ea commodo consequat. 
        Duis aute irure dolor in reprehenderit in voluptate velit 
        esse cillum dolore eu fugiat nulla pariatur. 
        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui 
        officia deserunt mollit anim id est laborum.`}
      </p>
      <p>
        {`Dictum non consectetur a erat nam at lectus. 
        In ante metus dictum at tempor commodo. 
        Quisque id diam vel quam elementum pulvinar 
        etiam non. Interdum velit euismod in 
        pellentesque massa placerat duis ultricies. 
        Erat velit scelerisque in dictum non. Tellus rutrum 
        tellus pellentesque eu tincidunt tortor aliquam. 
        Fringilla ut morbi tincidunt augue interdum velit. 
        A condimentum vitae sapien pellentesque 
        habitant morbi tristique senectus et. 
        Nunc aliquet bibendum enim facilisis gravida neque convallis a cras. 
        Lorem ipsum dolor sit amet consectetur. Integer malesuada 
        nunc vel risus commodo viverra maecenas accumsan lacus. 
        Arcu cursus euismod quis viverra nibh cras pulvinar.`}
      </p>
    </div>
  </div>
);

export default Home;
