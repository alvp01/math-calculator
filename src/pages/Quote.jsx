import React from 'react';

const Quote = () => (
  <div>
    <p>What is mathematics? It is only a systematic effort of solving puzzles posed by nature.</p>
    <p><strong>â€” Shakuntala Devi</strong></p>
  </div>
);

export default Quote;
