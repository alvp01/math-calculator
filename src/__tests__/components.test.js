import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Quote from '../pages/Quote';
import Home from '../pages/Home';
import Calculator from '../components/Calculator';

describe('Test home', ()=>{
  test('Home snapshot', ()=>{
    const tree = render(<Home />);
    expect(tree).toMatchSnapshot();
  });
});
