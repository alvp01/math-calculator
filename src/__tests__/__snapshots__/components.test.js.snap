// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test home Home snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="contentContainer"
      >
        <h1>
          Welcome to our page!
        </h1>
        <div
          class="paragraphs"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
        incididunt ut labore et dolore magna aliqua. 
        Ut enim ad minim veniam, quis nostrud exercitation ullamco 
        laboris nisi ut aliquip ex ea commodo consequat. 
        Duis aute irure dolor in reprehenderit in voluptate velit 
        esse cillum dolore eu fugiat nulla pariatur. 
        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui 
        officia deserunt mollit anim id est laborum.
          </p>
          <p>
            Dictum non consectetur a erat nam at lectus. 
        In ante metus dictum at tempor commodo. 
        Quisque id diam vel quam elementum pulvinar 
        etiam non. Interdum velit euismod in 
        pellentesque massa placerat duis ultricies. 
        Erat velit scelerisque in dictum non. Tellus rutrum 
        tellus pellentesque eu tincidunt tortor aliquam. 
        Fringilla ut morbi tincidunt augue interdum velit. 
        A condimentum vitae sapien pellentesque 
        habitant morbi tristique senectus et. 
        Nunc aliquet bibendum enim facilisis gravida neque convallis a cras. 
        Lorem ipsum dolor sit amet consectetur. Integer malesuada 
        nunc vel risus commodo viverra maecenas accumsan lacus. 
        Arcu cursus euismod quis viverra nibh cras pulvinar.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="contentContainer"
    >
      <h1>
        Welcome to our page!
      </h1>
      <div
        class="paragraphs"
      >
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor 
        incididunt ut labore et dolore magna aliqua. 
        Ut enim ad minim veniam, quis nostrud exercitation ullamco 
        laboris nisi ut aliquip ex ea commodo consequat. 
        Duis aute irure dolor in reprehenderit in voluptate velit 
        esse cillum dolore eu fugiat nulla pariatur. 
        Excepteur sint occaecat cupidatat non proident, sunt in culpa qui 
        officia deserunt mollit anim id est laborum.
        </p>
        <p>
          Dictum non consectetur a erat nam at lectus. 
        In ante metus dictum at tempor commodo. 
        Quisque id diam vel quam elementum pulvinar 
        etiam non. Interdum velit euismod in 
        pellentesque massa placerat duis ultricies. 
        Erat velit scelerisque in dictum non. Tellus rutrum 
        tellus pellentesque eu tincidunt tortor aliquam. 
        Fringilla ut morbi tincidunt augue interdum velit. 
        A condimentum vitae sapien pellentesque 
        habitant morbi tristique senectus et. 
        Nunc aliquet bibendum enim facilisis gravida neque convallis a cras. 
        Lorem ipsum dolor sit amet consectetur. Integer malesuada 
        nunc vel risus commodo viverra maecenas accumsan lacus. 
        Arcu cursus euismod quis viverra nibh cras pulvinar.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
